federatedscope/core/evaluator.py

```diff
@@ -38,7 +38,7 @@ def eval(self, ctx):
         results = {}
         y_true, y_pred, y_prob = self._check_and_parse(ctx)
         for metric, func in self.eval_metric.items():
-            results["{}_{}".format(ctx.cur_data_split,
+            results["{}_{}".format(ctx.cur_mode,
```

Comment:
I think the evaluator is performed on specific data split instead of the current mode.

Reply:
The `evaluator` class names the evaluation results during training/test/validate process. Supposing we are testing on the training data, it may be confused that the evaluation results are printed out by "train_avg_loss". 
Maybe we can print the chosen dataset (`cur_data_split`) before running the routine as part of the settings.

Reply:
Agree.